ingest/src/main/java/com/microsoft/azure/kusto/ingest/source/SourceInfo.java

```diff
@@ -1,17 +1,5 @@
 package com.microsoft.azure.kusto.ingest.source;
 
-import java.util.UUID;
-
-abstract class SourceInfo {
-
-    private UUID sourceId;
-
-    public UUID getSourceId() {
-        return sourceId;
-    }
-
-    public void setSourceId(UUID sourceId) {
-        this.sourceId = sourceId;
-    }
-
+public interface SourceInfo {
+    void validate();
```

Comment:
Guid Id

Reply:
@rabee333 I think the meaning here is that the interface should have the get/set for the sourceId and the abstract class will have the implementation (very basic) as well as the actual field